shell/platform/darwin/macos/framework/Source/FlutterMenuPlugin.mm

```diff
@@ -0,0 +1,418 @@
+//    action.
+// 4) ⌃⇧a is equivalent to ⌃A: If a keyEquivalent is a capitalized alphabetical
+//    letter and keyEquivalentModifierMask does not include
+//    NSEventModifierFlagShift, AppKit will add ⇧ automatically in the UI.
+
+/*
+ * Maps the string used by NSMenuItem for the given special key equivalent.
+ * Keys are the logical key ids of matching trigger keys.
+ */
+static NSDictionary<NSNumber*, NSNumber*>* GetMacOsSpecialKeys() {
+  return @{
+    @0x00100000008 : [NSNumber numberWithInt:NSBackspaceCharacter],
+    @0x00100000009 : [NSNumber numberWithInt:NSTabCharacter],
+    @0x0010000000a : [NSNumber numberWithInt:NSNewlineCharacter],
+    @0x0010000000c : [NSNumber numberWithInt:NSFormFeedCharacter],
+    @0x0010000000d : [NSNumber numberWithInt:NSCarriageReturnCharacter],
+    @0x0010000007f : [NSNumber numberWithInt:NSDeleteCharacter],
+    @0x00100000801 : [NSNumber numberWithInt:NSF1FunctionKey],
+    @0x00100000802 : [NSNumber numberWithInt:NSF2FunctionKey],
+    @0x00100000803 : [NSNumber numberWithInt:NSF3FunctionKey],
+    @0x00100000804 : [NSNumber numberWithInt:NSF4FunctionKey],
+    @0x00100000805 : [NSNumber numberWithInt:NSF5FunctionKey],
+    @0x00100000806 : [NSNumber numberWithInt:NSF6FunctionKey],
+    @0x00100000807 : [NSNumber numberWithInt:NSF7FunctionKey],
+    @0x00100000808 : [NSNumber numberWithInt:NSF8FunctionKey],
+    @0x00100000809 : [NSNumber numberWithInt:NSF9FunctionKey],
+    @0x0010000080a : [NSNumber numberWithInt:NSF10FunctionKey],
+    @0x0010000080b : [NSNumber numberWithInt:NSF11FunctionKey],
+    @0x0010000080c : [NSNumber numberWithInt:NSF12FunctionKey],
+    @0x0010000080d : [NSNumber numberWithInt:NSF13FunctionKey],
+    @0x0010000080e : [NSNumber numberWithInt:NSF14FunctionKey],
+    @0x0010000080f : [NSNumber numberWithInt:NSF15FunctionKey],
+    @0x00100000810 : [NSNumber numberWithInt:NSF16FunctionKey],
+    @0x00100000811 : [NSNumber numberWithInt:NSF17FunctionKey],
+    @0x00100000812 : [NSNumber numberWithInt:NSF18FunctionKey],
+    @0x00100000813 : [NSNumber numberWithInt:NSF19FunctionKey],
+    @0x00100000814 : [NSNumber numberWithInt:NSF20FunctionKey],
+
+    // For some reason, there don't appear to be constants for these in ObjC. In
+    // Swift, there is a class with static members for these: KeyEquivalent. The
+    // values below are taken from that (where they don't already appear above).
+    @0x00100000302 : @0xf702,  // ArrowLeft
+    @0x00100000303 : @0xf703,  // ArrowRight
+    @0x00100000304 : @0xf700,  // ArrowUp
+    @0x00100000301 : @0xf701,  // ArrowDown
+    @0x00100000306 : @0xf729,  // Home
+    @0x00100000305 : @0xf72B,  // End
+    @0x00100000308 : @0xf72c,  // PageUp
+    @0x00100000307 : @0xf72d,  // PageDown
+    @0x0010000001b : @0x001B,  // Escape
+  };
+}
+
+/*
+ * The mapping from the PlatformProvidedMenu enum to the macOS selectors for the provided
+ * menus.
+ */
+static const std::map<int, SEL> GetMacOSProvidedMenus() {
```

Comment:
`std::map<flutter::PlatformProvidedMenu, SEL>`

(and tack on the `flutter::` namespace for the keys below once the enum is namespaced)

Reply:
Done.